ntoskrnl/fsrtl/unc.c

```diff
@@ -71,7 +71,7 @@ FsRtlpIsDfsEnabled(VOID)
         return TRUE;
     }
 
-    return ((ULONG)KeyQueryOutput.KeyInfo.Data != 1);
+    return ((*(PULONG)KeyQueryOutput.KeyInfo.Data) != 1);
```

Comment:
Unless I'm missing something, this is not functionally equivalent.

Reply:
Yes, it is a bug-fix!